### PR TITLE
Added Types methods to Protocol

### PR DESCRIPTION
**Pull request by EliaBracciSumo**
Added method `Types` to return Protocol `Types`.
Added method `OriginalString` to return Protocol original schema

### CODE REVIEW

**Review by nrwiersma — APPROVED** [review]
LGTM 🎉

File: protocol.go [hunk 1]
```diff
@@ -49,13 +49,16 @@ type Protocol struct {
...
+
+	originalString string
 }
 
 // NewProtocol creates a protocol instance.
 func NewProtocol(
 	name, namepsace string,
 	types []NamedSchema,
 	messages map[string]*Message,
+	protocolString string,
```
Comment (nrwiersma) [inline protocol.go:61]: This is a breaking change. It cannot be added until the next major

**Review by nrwiersma — CHANGES_REQUESTED** [review]

**Comment by EliaBracciSumo** [comment]
@nrwiersma thank you for your review 🙏 
However, I would like to add also another field in Protocol struct.

I called it `OriginalString` but we can change it. The intent is to save the string of the protocol as is, using its non-canonical form. WDYT?

I need this to generate Protocol hash

File: protocol.go (hunk 1 above)
Comment (EliaBracciSumo) [reply protocol.go:61]: Let me create two different PRs. Or let me find a way to avoid this new parameter

Comment (EliaBracciSumo) [reply protocol.go:61]: https://github.com/hamba/avro/pull/315#issuecomment-1772860065 added a comment here in the meantime

**Comment by nrwiersma** [comment]
You can open an issue or discussion about `OriginalString`.

**Review by nrwiersma — APPROVED** [review]

File: protocol.go (hunk 1 above)
Comment (EliaBracciSumo) [reply protocol.go]: Moving discussion in another PR https://github.com/hamba/avro/pull/316/files

**Comment by EliaBracciSumo** [comment]
@nrwiersma when will it be released?

**Comment by nrwiersma** [comment]
I usually wait a coupe days for any other changes that come in, to group the release. It seems no more will come in, so in a moment.

**Pull request merged**